### PR TITLE
People object changed to no longer require fields not needed for adaptive comfort models 

### PR DESCRIPTION
**Pull request by lgu1234**
Pull request overview
---------------------
Allow adaptive thermal models in People objects to work with irrelevant blank fields.

addressess Issue #5948  

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [x] Code style (parentheses padding, variable names)
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check -- verify this
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [ ] Verify naming conventions and styles, memos and notes and defaults
   - [ ] Open windows IDF Editor with modified IDD to check for errors
   - [ ] If transition, add rules to spreadsheet
   - [ ] If transition, add transition source
   - [ ] If transition, update idfs
 - [ ] If new idf included, locally check the err file and other outputs
 - [x] Documentation changes in place
 - [ ] Changed docs build successfully
 - [ ] ExpandObjects changes?
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

**Comment by EnergyArchmage** [comment]
I/O ref description of People object should also get some modifications, with the same content as in the new IDD \note. Otherwise this is  looking good to go.

File: idd/Energy+.idd.in [hunk 1]
```diff
@@ -19678,11 +19678,11 @@ People,
        \minimum 0.0
        \maximum 1.0
   A5 , \field Activity Level Schedule Name
-       \required-field
        \note Note that W has to be converted to mets in TC routine
        \type object-list
        \note units in schedule are W/person
        \object-list ScheduleNames
+       \note optional (only required for runs of thermal comfort models: Fanger, Pierce and KSU)
```
Comment (mjwitte) [inline idd/Energy+.idd.in]: Not so.  Activity level is used in the zone heat balance.  This should remain required.

Comment (lgu1234) [reply idd/Energy+.idd.in]: @mjwitte I am going to set it back as required.

Comment (mjwitte) [reply idd/Energy+.idd.in]: And remove the new note, also.

**Comment by lgu1234** [comment]
@EnergyArchmage I am going to revise People object portion of IO Ref. Thanks.

File: idd/Energy+.idd.in (hunk 1 above)
Comment (lgu1234) [reply idd/Energy+.idd.in]: @mjwitte  I already did. Thanks.

**Comment by lgu1234** [comment]
@EnergyArchmage @mjwitte The required field was revoked. The corresponding changes have been uploaded for further review and comments: source code, unit test, idd and IO Ref.

**Comment by EnergyArchmage** [comment]
@lgu1234 Okay, I have reviewed this in more detail.  I have made some code corrections, seems the logic change is needed in these places as well, please double check.  I have also edited the I/O ref content some and added missing discussion of comfort angle factor list. 
https://github.com/NREL/EnergyPlus/blob/137996517-Remove-Adaptive-comfort-restriction-from-irrelevant-fields/src/EnergyPlus/InternalHeatGains.cc#L685

https://github.com/NREL/EnergyPlus/blob/137996517-Remove-Adaptive-comfort-restriction-from-irrelevant-fields/src/EnergyPlus/ThermalComfort.cc#L1940

**Comment by lgu1234** [comment]
@EnergyArchmage Thanks for the changes. I like them.

**Comment by EnergyArchmage** [comment]
This is ready to go in now.

**Comment by Myoldmopar** [comment]
@nrel-bot is taking forever, CI looks very happy so far though.  Merging.

**Pull request merged**